src/commands.ts

```diff
@@ -0,0 +1,386 @@
+            execute: async () => hostsCMD(shell)
+        },
+        home: {
+            name: "home",
+            help: "Go to the home page",
+            usage: "home",
+            execute: async () => terminal7.goHome()
+        }
+    }))
+}
+
+async function helpCMD(shell: Shell, args: string[]) {
+    let help = ""
+    if (!args[0]) {
+        help += "\x1B[1mAvailable commands:\x1B[0m\n"
+        for (const [name, command] of shell.commands) {
+            help += `  ${name}: ${command.help}\n`
+        }
+        help += "\nType 'help <command>' for more information."
+    } else {
+        const command = shell.commands.get(args[0])
+        if (!command)
+            help = `Command not found: ${args[0]}`
+        else {
+            help += `\x1B[1m${command.name}\x1B[0m\n`
+            help += `  ${command.help}\n`
+            help += `  Usage: ${command.usage}`
+        }
+    }
+    shell.t.writeln(help)
+}
+
+async function fortuneCMD(shell: Shell) {
+    const res = await fetch("https://raw.githubusercontent.com/ruanyf/fortunes/master/data/fortunes")
+    shell.t.writeln((await res.text()).split("%\n")[Math.floor(Math.random() * 100)].trim())
+}
+
+async function echoCMD(shell: Shell, args: string[]) {
+    shell.t.writeln(args.join(" "))
+}
+
+async function connectCMD(shell:Shell, args: string[]) {
+    const hostname = args[0]
+    if (!hostname)
+        return shell.t.writeln("Missing hostname")
+    const gates = terminal7.gates
+    const gate = gates.get(hostname)
+    if (!gate)
+        return shell.t.writeln(`Host not found: ${hostname}`)
+    await new Promise<void>((resolve) => {
+        gate.connect(() => {
+            gate.load()
+            resolve()
+        }, () => {
+            shell.t.writeln(`Failed to connect to ${hostname}`)
+            resolve()
+        })
+    })
+}
+
+async function addHostCMD(shell: Shell) {
+    if (!terminal7.conf.peerbook) {
+        const pbForm = [
+            { prompt: "Add static host" },
+            { prompt: "Setup peerbook" }
+        ]
+        let choice
+        try {
+            choice = await shell.newForm(pbForm, "menu")
+        } catch (e) {
+            return
+        }
+        if (choice == "Setup peerbook") {
+            await peerbookForm(shell)
+            return
+        }
+    }
+    const f = [
+        { prompt: "Enter destination (ip or domain)" }
+    ]
+    let hostname
+    try {
+        hostname = (await shell.newForm(f, "text"))[0]
+    } catch (e) { 
+        return
+    }
+
+    if (terminal7.validateHostAddress(hostname)) {
+        shell.t.writeln(`  ${hostname} already exists, connecting...`)
+        terminal7.gates.get(hostname).connect()
+        return
+    }
+    terminal7.activeG = terminal7.addGate({
+        name: "temp_" + Math.random().toString(16).slice(2), // temp random name
+        addr: hostname,
+        id: hostname
+    }, false)
+    shell.map.refresh()
+    await terminal7.activeG.CLIConnect(() => {
```

Comment:
I don't like the `CLIConnect` method as now that we have commands and shell they should hold all the interaction and leave the logic to gate.